File: src/gazap-site/src/main/java/gazap/site/validation/UserEmailValidator.java
package gazap.site.validation;

import org.springframework.util.StringUtils;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

public class UserEmailValidator implements ConstraintValidator<UserEmail, String> {
    org.apache.commons.validator.routines.EmailValidator emailValidator = org.apache.commons.validator.routines.EmailValidator.getInstance();

    @Override
    public void initialize(UserEmail constraintAnnotation) {
    }

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        return !StringUtils.hasLength(value) || emailValidator.isValid(value);
    }
}

File: src/waypalm-site/src/main/java/waypalm/site/validation/UserEmailValidator.kt
package waypalm.site.validation

import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext
import org.springframework.util.StringUtils

public class UserEmailValidator
: ConstraintValidator<UserEmail, String>
{
    val emailValidator: org.apache.commons.validator.routines.EmailValidator = org.apache.commons.validator.routines.EmailValidator.getInstance()!!;

    public override fun initialize(constraintAnnotation: UserEmail?) = Unit.VALUE

    public override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean {
        return !StringUtils.hasLength(value) || emailValidator.isValid(value);
    }
}